ikarus/finiteElements/mechanics/nonLinearElasticityFE.hh

```diff
@@ -114,26 +143,134 @@ namespace Ikarus {
           const auto u = uFunction.evaluate(quadPos);
 
           // Value of the Neumann data at the current position
-          auto neumannValue = neumannBoundaryLoad_(toEigen(intersection.geometry().global(curQuad.position())), lambda);
-
+          auto neumannValue
+              = (*neumannBoundaryLoad_)(toEigen(intersection.geometry().global(curQuad.position())), lambda);
           energy -= neumannValue.dot(u) * curQuad.weight() * integrationElement;
         }
       }
-
       return energy;
     }
 
+    void calculateMatrix(const FERequirementType& par, typename Traits::MatrixType& K) const {
+      using namespace Dune::DerivativeDirections;
+      using namespace Dune;
+      const auto eps = getStrainFunction(par);
+      const auto geo = this->localView().element().geometry();
+
+      for (const auto& [gpIndex, gp] : eps.viewOverIntegrationPoints()) {
+        const double intElement = geo.integrationElement(gp.position()) * gp.weight();
+        const auto EVoigt       = (eps.evaluate(gpIndex, on(Dune::DerivativeDirections::gridElement))).eval();
+        auto C                  = getMaterialTangent(EVoigt);
+        auto stresses           = getStress(EVoigt);
+        for (size_t i = 0; i < numberOfNodes; ++i) {
+          const auto bopI = eps.evaluateDerivative(gpIndex, wrt(coeff(i)), on(Dune::DerivativeDirections::gridElement));
+          for (size_t j = 0; j < numberOfNodes; ++j) {
+            const auto bopJ
+                = eps.evaluateDerivative(gpIndex, wrt(coeff(j)), on(Dune::DerivativeDirections::gridElement));
+            const auto bopIJ = eps.evaluateDerivative(gpIndex, wrt(coeff(i, j)), along(stresses),
+                                                      on(Dune::DerivativeDirections::gridElement));
+            K.template block<myDim, myDim>(i * myDim, j * myDim) += (bopI.transpose() * C * bopJ + bopIJ) * intElement;
+          }
+        }
+      }
+    }
+
+    void calculateAt(const ResultRequirementsType& req, const Eigen::Vector<double, Traits::mydim>& local,
+                     ResultTypeMap<double>& result) const {
+      using namespace Dune::Indices;
+      using namespace Dune::DerivativeDirections;
+      using namespace Dune;
+
+      const auto eps    = getStrainFunction(req.getFERequirements());
+      auto gp           = toDune(local);
+      const auto EVoigt = (eps.evaluate(gp, on(gridElement))).eval();
+      auto C            = getMaterialTangent(EVoigt);
+
+      auto secondPiolaKirchhoffStress = (C * EVoigt).eval();

```

Comment:
this is only correct for St. Venant Kirchhoff. You should call `getStress` here. Furthermore the calculation souhld be inside the if statement. Otherwise the calculation is not needed.